src/app/src/main/kotlin/org/vaccineimpact/api/app/repositories/jooq/JooqBurdenEstimateRepository.kt

```diff
@@ -221,58 +219,80 @@ class JooqBurdenEstimateRepository(
         return mapper.mapBurdenEstimateSets(records)
     }
 
-    override fun getBurdenEstimateOutcomesSequence(groupId: String, touchstoneVersionId: String, scenarioId: String, burdenEstimateSetId: Int)
-            : Sequence<BurdenEstimateOutcome>
+    override fun getBurdenEstimateOutcomesSequence(burdenEstimateSetId: Int,
+                                                   outcomes: List<Pair<Short, String>>,
+                                                   disease: String)
+            : Sequence<BurdenEstimate>
     {
-        //check that the burden estimate set exists in the group etc
-        val set = getBurdenEstimateSet(groupId, touchstoneVersionId, scenarioId, burdenEstimateSetId)
 
-        if (set.isStochastic())
-        {
-            throw InvalidOperationError("Cannot get burden estimate data for stochastic burden estimate sets")
-        }
+        val countryLookup = getCountriesAsLookup()
 
-        return dsl.select(
-                DISEASE.ID,
-                BURDEN_ESTIMATE.YEAR,
-                BURDEN_ESTIMATE.AGE,
-                COUNTRY.ID,
-                COUNTRY.NAME,
-                BURDEN_OUTCOME.CODE,
-                BURDEN_ESTIMATE.VALUE
-        )
-                .fromJoinPath(BURDEN_ESTIMATE_SET, BURDEN_ESTIMATE)
-                .join(RESPONSIBILITY)
-                .on(BURDEN_ESTIMATE_SET.RESPONSIBILITY.eq(RESPONSIBILITY.ID))
-                .joinPath(RESPONSIBILITY, SCENARIO, SCENARIO_DESCRIPTION, DISEASE)
-                .join(COUNTRY)
-                .on(BURDEN_ESTIMATE.COUNTRY.eq(COUNTRY.NID))
-                .joinPath(BURDEN_ESTIMATE, BURDEN_OUTCOME)
-                .where(BURDEN_ESTIMATE_SET.ID.eq(burdenEstimateSetId))
-                .orderBy(
-                        DISEASE.ID,
-                        BURDEN_ESTIMATE.YEAR,
-                        BURDEN_ESTIMATE.AGE,
-                        COUNTRY.ID,
-                        COUNTRY.NAME,
-                        BURDEN_OUTCOME.CODE)
-                .fetchSequence()
-                .map { it.into(BurdenEstimateOutcome::class.java) }
+        val cohortSize = dsl.select(BURDEN_OUTCOME.ID)
+                .from(BURDEN_OUTCOME)
+                .where(BURDEN_OUTCOME.CODE.eq("cohort_size"))
+                .fetchAnyInto(Short::class.java)
+
+        return countryLookup.keys.asSequence().map { k ->
+            val cursor = dsl.fetchLazy(getBurdenEstimateQueryForCountry(burdenEstimateSetId, outcomes, cohortSize, k))
+            generateSequence {
+                cursor.fetchNext()?.map { record ->
+                    val country = countryLookup.getValue(record.get("country", Short::class.java))
+                    val outcomeList = outcomes.filter { it.second != "cohort_size" }
```

Comment:
Cohort size is treated as an outcome for burden estimate sets? Could do this filter outside the lambda I guess.

Reply:
(yes cohort_size is an outcome)

Reply:
Oh, although actually cohort_size shouldn't be in the expectations table, so this isn't needed here